linker/Tests/TestCasesRunner/TestRunner.cs

```diff
@@ -93,5 +94,22 @@ private LinkedTestCaseResult Link (TestCase testCase, TestCaseSandbox sandbox, M
 		protected virtual void AddAdditionalLinkOptions (LinkerArgumentBuilder builder, TestCaseMetadaProvider metadataProvider)
 		{
 		}
+
+		private T GetResultOfTaskThatMakesNUnitAssertions<T> (Task<T> task)
+		{
+			try {
+				return task.Result;
+			} catch (AggregateException e) {
+				if (e.InnerException != null) {
```

Comment:
Why not to check only for `ResultStateException` ?

Reply:
We use an older version of nunit and ResultStateException doesn't exist in our version.  Checking for each exception is a little extra code, but it's more backward compatible with older nunit versions